src/test/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialBindingTest.java

```diff
@@ -0,0 +1,109 @@
+package org.jenkinsci.plugins.kubernetes.credentials;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import hudson.FilePath;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.*;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.codehaus.groovy.runtime.DefaultGroovyMethods.grep;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class OpenShiftBearerTokenCredentialBindingTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    private Server server;
+    private String serverUri;
+
+    @Before
+    public void prepareFakeOAuthServer() throws Exception {
```

Comment:
Again, I don't know anything about `RestartableJenkinsRule`, but if you add some log traces you can see that `@After` is executed before the story actually runs. This means your mock server will be stopped before your Workflow job executes. I believe that's why you tried to save `serverUri` in a shared variable

Not sure if it's the right thing to do, but starting the mock server as a first step solves the loading issue.
```
        story.addStep(new Statement() {
            @Override public void evaluate() throws Throwable {
                ServletContextHandler context = new ServletContextHandler();
                context.setContextPath("/");
                context.addServlet(new ServletHolder(new MockHttpServlet()), "/*");

                server = new Server(0);
                server.setHandler(context);
                server.start();
            }
        });
```